pkg/envoy/xds_server.go

```diff
@@ -237,7 +242,7 @@ func (s *xdsServer) start() error {
 
 	resourceConfig := s.initializeXdsConfigs()
 
-	s.stopFunc = startXDSGRPCServer(socketListener, resourceConfig)
+	s.stopFunc = s.startXDSGRPCServer(socketListener, resourceConfig)
```

Comment:
Looking at the refactored version, i start to wonder whether it would actually be possible to wait with the start of the xDS gRPC server until restoration is finished. Might be cleaner instead of putting the wait logic into all the observers :thinking: 

But definitely not blocking - just wondering whether you also checked this variant?

Reply:
Thought about this, but this allows Envoy to connect to the new agent instance sooner. There are already a bunch of retries for reconnecting the xDS streams in Envoy logs, and while not strictly harmful, it seems better to me that that Envoy can connect and be ready to receive the (hopefully unchanged) resources ASAP.

Reply:
Makes sense. Thanks!